### PR TITLE
add tutorial for Reporter and report.

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
related to #3290.

The contents will be:
- What is Reporter?
- Use report in Chain or Link
- Naming rule for the reported values

### CODE REVIEW

File: docs/source/tutorial/report.rst [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+If these values are reported inside the :class:`~chainer.training.extensions.Evaluator`
+extension, ``'validation/'`` is added at the head of the link name, thus
+the item names are changed to ``'validation/main/loss'`` and ``'validation/main/accuracy'``
+(``'validation'`` is the default name of the Evaluator extension).
+
+Naming rule for the reported values
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+So, you have known almost everything about :class:`~chainer.Reporter`.
+However, one content is remained. It is what is the naming rule for the reported values,
```
Comment (bkvogel) [inline docs/source/tutorial/report.rst]: suggested wording change:
So, you know almost everything about :class:`~chainer.Reporter`.
 However, there is one more thing.

File: docs/source/tutorial/report.rst [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+especially when the values are reported from links that is not the root of the link hierarchy.
```
Comment (bkvogel) [inline docs/source/tutorial/report.rst]: from links that is -> from a link that is

File: docs/source/tutorial/report.rst [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+As we explained in the previous section, the root of links is named as ``'main'`` 
+by the the :class:`~chainer.training.StandardUpdater` and the names of reported
+values in the root have the prefix ``'main/'``.
+When the values are reported from links that is not the root of the link hierarchy,
```
Comment (bkvogel) [inline docs/source/tutorial/report.rst]: from links that is not the root -> from a link that is not the root

**Comment by bkvogel** [comment]
Thank you for the PR and apologies for the delay in reviewing! I think it looks good. If you can fix the minor wording issues and resolve the branch conflicts, I think it will be good.

**Comment by keisuke-umezawa** [comment]
@bkvogel Thank you for the review. I fix them. 

Recently, the documentation directories are re-organized, and I moved the file in docs/source/guides.
Is it okay?

**Comment by keisuke-umezawa** [comment]
something like this.

**Comment by bkvogel** [comment]
Sure, I think it's fine.

**Comment by bkvogel** [comment]
Jenkins, test this please.

**Comment by bkvogel** [comment]
It seems the doctest failed. Could you check. I am not sure, but maybe you need to add another `# doctest: +SKIP`?

File "guides/report.rst", line 155, in default
16:35:05 Failed example:
16:35:05     optimizer.setup(model)
16:35:05 Expected nothing
16:35:05 Got:
16:35:05     <chainer.optimizers.sgd.SGD object at 0x7f3d179f8278>
16:35:05 
16:35:05 
16:35:08 make: *** [doctest] Error 1
16:35:08 Makefile:176: recipe for target 'doctest' failed

**Comment by bkvogel** [comment]
@keisuke-umezawa I Think it looks good other than the doctest error. If you can fix the error, I will then merge.

**Comment by keisuke-umezawa** [comment]
@bkvogel Thanks! I will fix.

**Comment by bkvogel** [comment]
Jenkins, test this please.

**Comment by bkvogel** [comment]
LGTM!

**Pull request merged**